cpp/src/copying/scatter.cu

```diff
@@ -63,21 +116,30 @@ void scatter(table const* source_table, gdf_index_type const scatter_map[],
     return;
   }
   
-  CUDF_EXPECTS(nullptr != scatter_map, "scatter_map is null");
+  gdf_column gather_map = cudf::allocate_like(scatter_map, num_destination_rows);
 
-  constexpr gdf_index_type default_index_value = -1;  
-  rmm::device_vector<gdf_index_type> v_gather_map(num_destination_rows, default_index_value);
- 
-  constexpr int block_size = 256;
-
-  const gdf_size_type invert_grid_size =
-    (destination_table->num_rows() + block_size - 1) / block_size;
+  type_dispatcher(scatter_map.dtype, invert_map_kernel_forwarder{},
+		  gather_map,
+		  num_destination_rows,
+		  scatter_map,
+		  num_source_rows);
 
-  detail::invert_map<<<invert_grid_size, block_size>>>(v_gather_map.data().get(), num_destination_rows, scatter_map, num_source_rows);
-  
-  // We want to check bounds for scatter since it is possible that
+  // We want to ignore out of bounds indices for scatter since it is possible that
   // some elements of the destination column are not modified. 
-  detail::gather(source_table, v_gather_map.data().get(), destination_table, true, true);    
+  detail::gather(source_table, gather_map, destination_table, false, true, true);
+
+  gdf_column_free(&gather_map);
+}
+
+void scatter(table const* source_table, gdf_index_type const scatter_map[],
```

Comment:
I thought you were going to adds bounds checking to `scatter` as well?

Reply:
It's been added